lib/arrow/shuttle/shape.ex

```diff
@@ -23,8 +23,33 @@ defmodule Arrow.Shuttle.Shape do
   @doc false
   def changeset(shape, attrs) do
     shape
-    |> cast(attrs, [:name, :path, :bucket, :prefix])
+    |> cast(attrs, [:name, :path, :bucket, :prefix], empty_values: ["-S"] ++ empty_values())
     |> validate_required([:name, :path, :bucket, :prefix])
     |> unique_constraint(:name)
+    |> validate_change(:name, fn :name, name ->
+      if String.ends_with?(name, "-S") do
+        []
+      else
+        [{:name, "must end with -S"}]
+      end
+    end)
+  end
+
+  def validate_and_enforce_name(attrs) do
+    changeset =
+      %__MODULE__{}
+      |> cast(attrs, [:name])
+      |> validate_required([:name])
+
+    cond do
+      !changeset.valid? ->
```

Comment:
nit: Assuming `changeset.valid?` always has a boolean value, prefer using `not`
```suggestion
      not changeset.valid? ->
```